cmd/clusterctl/client/config/provider.go

```diff
@@ -87,3 +90,22 @@ func NewProvider(name string, url string, ttype clusterctlv1.ProviderType) Provi
 		providerType: ttype,
 	}
 }
+
+func (p provider) MarshalJSON() ([]byte, error) {
+	dir, file := filepath.Split(p.url)
+	j, err := json.Marshal(struct {
+		Name         string
+		ProviderType clusterctlv1.ProviderType
+		URL          string
+		File         string
```

Comment:
@wfernandes  Although I have struct In order of,
```
Name:
ProviderType:
URL:
File:
```
but the output is in the form,

```
File:
Name:
ProviderType:
URL:
```
I am not sure how to fix this. If you can help. Thanks

Reply:
Yeah, it's probably go-yaml sorting alphabetically. I'm not too concerned about this since the point of the YAML output is so that other tools can consume and parse it easily. The humans can read the text output 🙂